### PR TITLE
Fix flaky timestamp validation in unit tests

### PR DESCRIPTION
**Pull request by konstantin-msft**
This pull request improves the reliability of authentication result tests by addressing timing issues caused by clock drift, and introduces a utility for tolerant result comparison. It also updates tests to use this utility, ensuring that tests are not flaky due to second-boundary timing differences. Additionally, it enhances test setup practices to prevent time-related race conditions.

**Test stability and reliability improvements:**

* Added `expectAuthenticationResult` utility in `StringConstants.ts` to compare `AuthenticationResult` objects while tolerating small differences in expiry timestamps, reducing test flakiness caused by timing differences.
* Updated all relevant test cases in `PopupClient.spec.ts`, `RedirectClient.spec.ts`, `SilentIframeClient.spec.ts`, and `Authorize.spec.ts` to use `expectAuthenticationResult` instead of direct equality checks, and refactored tests to reuse a single expected result object where possible. [[1]](diffhunk://#diff-0e4f86d8a16dd8be09b5f3b33d82dafe4c4325e3fd92b994120b0290beb0c2d6L1036-R1040) [[2]](diffhunk://#diff-6892d11e0bf0f9de499d836141fe1b0e7fe40f2b7d1d28e57830a083ba0080a4L3502-R3506) [[3]](diffhunk://#diff-ee3bc256edbe0cc9b07264ac953b539bf0be0c3737eb24a91cf11b35e20086a3L1807-R1811) [[4]](diffhunk://#diff-383f979b9a05a2d7fe02052138e8087f6ca2167e78d44dc4d41c391463d4da04L389-R400)

**Test utility and fixture improvements:**

* Modified `getTestAuthenticationResult` to read the clock only once per invocation, ensuring consistent expiry timestamps within the same result object. [[1]](diffhunk://#diff-2e5c268e977ace942971e41e81ebc723555fe267f5389c84bcc6caaad7f58497R563-R569) [[2]](diffhunk://#diff-2e5c268e977ace942971e41e81ebc723555fe267f5389c84bcc6caaad7f58497L572-R586)

**Test setup enhancements:**

* Updated `PopTokenGenerator.spec.ts` to freeze the clock before each test, preventing timing-related race conditions in tests that depend on the current time.

**Test assertion improvements:**

* Updated the `CacheManager.spec.ts` test to assert the presence and type of the `lastUpdatedAt` field rather than its exact value, accounting for the use of different clock reads during test execution.

### CODE REVIEW

File: lib/msal-browser/test/utils/StringConstants.ts [hunk 1]
```diff
@@ -591,6 +592,64 @@ export function getTestAuthenticationResult(): AuthenticationResult {
...
+    } = actual as AuthenticationResult;
+    const {
+        expiresOn: expectedExpiresOn,
+        extExpiresOn: expectedExtExpiresOn,
+        ...expectedRest
+    } = expected;
+
+    expect(actualRest).toEqual(expectedRest);
+
+    const withinTolerance = (
```
Comment (tnorling) [inline lib/msal-browser/test/utils/StringConstants.ts]: This problem is already solved by [Jest's useFakeTimers API](https://jestjs.io/docs/jest-object#fake-timers). We shouldn't need this helper function at all. The fix for the flaky tests can just be to add a `useFakeTimers()` call in the `beforeEach` and `useRealTimers()` in the `afterEach` for any affected tests

Comment (konstantin-msft) [reply lib/msal-browser/test/utils/StringConstants.ts]: Great suggestion, thanks

Comment (konstantin-msft) [reply lib/msal-browser/test/utils/StringConstants.ts]: Removed the redundant helper entirely

**Review by lalimasharda — APPROVED** [review]

**Comment by hectormmg** [comment]
@konstantin-msft a while back I manually froze timestamps for a few tests, do you mind making sure those patches are updated/removed to match this approach?

**Review by sameerag — APPROVED** [review]

**Comment by konstantin-msft** [comment]
> Konstantin (@konstantin-msft) a while back I manually froze timestamps for a few tests, do you mind making sure those patches are updated/removed to match this approach?

The original patch that mocks `jest.spyOn(Date, "now").mockReturnValue(Date.now());` does not address `TimeUtils.nowSeconds()` and similar methods, which don't rely on `Date.now()`. I will create a follow-up PR to consolidate date freeze.
